pkg/scorecard_result.go

```diff
@@ -135,9 +155,105 @@ func (r *ScorecardResult) AsString(showDetails bool, writer io.Writer) error {
 	table.SetCenterSeparator("|")
 	table.AppendBulk(data)
 	table.Render()
+
+	return nil
+}
+
+// UPGRADEv2: new code.
+func (r *ScorecardResult) AsString2(showDetails bool, logLevel zapcore.Level, writer io.Writer) error {
+	sortedChecks := make([]checker.CheckResult, len(r.Checks))
+	//nolint
+	// UPGRADEv2: not needed after upgrade.
+	for i, checkResult := range r.Checks {
+		sortedChecks[i] = checkResult
+	}
+	sort.Slice(sortedChecks, func(i, j int) bool {
+		if sortedChecks[i].Pass == sortedChecks[j].Pass {
+			return sortedChecks[i].Name < sortedChecks[j].Name
+		}
+		return sortedChecks[i].Pass
+	})
+
+	data := make([][]string, len(sortedChecks))
+	//nolint
+	// UPGRADEv2: not needed after upgrade.
+	for i, row := range sortedChecks {
+		//nolint
+		if row.Version != 2 {
+			continue
+		}
+		const withdetails = 5
+		const withoutdetails = 4
+		var x []string
+
+		if showDetails {
+			x = make([]string, withdetails)
+		} else {
+			x = make([]string, withoutdetails)
+		}
+
+		// UPGRADEv2: rename variable.
+		if row.Score2 == checker.InconclusiveResultScore {
+			x[0] = "?"
+		} else {
+			x[0] = fmt.Sprintf("%d", row.Score2)
+		}
+
+		doc := fmt.Sprintf("https://github.com/ossf/scorecard/blob/main/checks/checks.md#%s", strings.ToLower(row.Name))
+		x[1] = row.Reason2
+		x[2] = row.Name
+		if showDetails {
+			// UPGRADEv2: change to make([]string, len(row.Details))
```

Comment:
make a helper for this, since it seems to be mostly the same as line 130~. Good to have this in a separate function anyway even after we delete the older code.

Reply:
agreed. I think I procrastinated this change :-)